prisma-fmt/src/text_document_completion/datasource.rs

```diff
@@ -0,0 +1,172 @@
+                r#"shadowDatabaseUrl = "String" | env("ENVIRONMENT_VARIABLE")"#,
+                r#"Connection URL including authentication info to use for Migrate's [shadow database](https://pris.ly/d/migrate-shadow)."#,
+                None,
+            ),
+        })),
+        ..Default::default()
+    })
+}
+
+pub(super) fn url_completion(completion_list: &mut CompletionList) {
+    completion_list.items.push(CompletionItem {
+        label: "engines url".to_owned(),
+        insert_text: Some(r#"url = $0"#.to_owned()),
+        insert_text_format: Some(InsertTextFormat::SNIPPET),
+        kind: Some(CompletionItemKind::FIELD),
+        documentation: Some(Documentation::MarkupContent(MarkupContent {
+            kind: MarkupKind::Markdown,
+            value: generate_pretty_doc(
+                r#"url = "String" | env("ENVIRONMENT_VARIABLE")"#,
+                r#"Connection URL including authentication info. Each datasource provider documents the URL syntax. Most providers use the syntax provided by the database. [Learn more](https://pris.ly/d/connection-strings)."#,
+                None,
+            ),
+        })),
+        ..Default::default()
+    })
+}
+
+pub(super) fn provider_completion(completion_list: &mut CompletionList) {
+    completion_list.items.push(CompletionItem {
+        label: "engines provider".to_owned(),
+        insert_text: Some(r#"provider = $0"#.to_owned()),
+        insert_text_format: Some(InsertTextFormat::SNIPPET),
+        kind: Some(CompletionItemKind::FIELD),
+        documentation: Some(Documentation::MarkupContent(MarkupContent {
+            kind: MarkupKind::Markdown,
+            value: generate_pretty_doc(
+                r#"provider = "foo""#,
+                r#"Describes which datasource connector to use. Can be one of the following datasource providers: `postgresql`, `mysql`, `sqlserver`, `sqlite`, `mongodb` or `cockroachdb`."#,
+                None,
+            ),
+        })),
+        ..Default::default()
+    })
+}
+
+pub(super) fn url_env_completion(completion_list: &mut CompletionList) {
+    completion_list.items.push(CompletionItem {
+        label: "engines env".to_owned(),
+        insert_text: Some(r#"env($0)"#.to_owned()),
+        insert_text_format: Some(InsertTextFormat::SNIPPET),
+        kind: Some(CompletionItemKind::PROPERTY),
+        documentation: Some(Documentation::MarkupContent(MarkupContent {
+            kind: MarkupKind::Markdown,
+            value: generate_pretty_doc(
+                r#"env(_ environmentVariable: string)"#,
+                r#"Specifies a datasource via an environment variable. When running a Prisma CLI command that needs the database connection URL (e.g. `prisma generate`), you need to make sure that the `DATABASE_URL` environment variable is set. One way to do so is by creating a `.env` file. Note that the file must be in the same directory as your schema.prisma file to automatically be picked up by the Prisma CLI.""#,
```

Comment:
lol, I think `generate` is pretty much the only one that does _not_ need it (besides format and validate and -v). 
Weird example we chose back in the day.
I would suggest replacing this with `db pull` and/or `db push`.

Reply:
Yeah, possibly! I haven't touched any of the completion docs, just ported over what was already there in language-tools, am happy to do an update at the same time too tho